packages/@aws-cdk/aws-apigateway/lib/authorizer.ts

```diff
@@ -1,14 +1,29 @@
-import { Resource } from '@aws-cdk/core';
+import { Construct, Resource } from '@aws-cdk/core';
 import { AuthorizationType } from './method';
 import { RestApi } from './restapi';
 
+const AUTHORIZER_SYMBOL = Symbol.for('@aws-cdk/aws-apigateway.Authorizer');
+
 /**
  * Base class for all custom authorizers
  */
 export abstract class Authorizer extends Resource implements IAuthorizer {
+  /**
+   * Return whether the given object is an Authorizer.
+   */
+  public static isAuthorizer(x: any): x is Authorizer {
+    return x !== null && typeof(x) === 'object' && AUTHORIZER_SYMBOL in x;
+  }
+
   public readonly abstract authorizerId: string;
   public readonly authorizationType?: AuthorizationType = AuthorizationType.CUSTOM;
 
+  constructor(scope: Construct, id: string) {
```

Comment:
```suggestion
  public constructor(scope: Construct, id: string) {
```

Reply:
Err, I think that's the default? 
I've not been specifying public anywhere and it has worked fine so far.